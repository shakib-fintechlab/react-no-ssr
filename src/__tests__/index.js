import React from 'react'

import NoSSR from '../'

describe('NoSSR', () => {
  describe('onServer', () => {
    describe('without onSSR components', () => {
      const MyComp = () => <div>Hello</div>
      const wrapper = render(
        <NoSSR>
          <MyComp />
        </NoSSR>
      )
      it('Should not render children', () => {
        // expect(wrapper.html()).toBeNull
        expect(wrapper).toMatchSnapshot()
      })
    })
    describe('with onSSR component', () => {
      const MyComp = () => <div>Hello</div>
      const Loading = () => <div>Loading...</div>
      const wrapper = render(
        <NoSSR onSSR={<Loading />}>
          <MyComp />
        </NoSSR>
      )
      it('Should show the onSSR component', () => {
        // expect(wrapper.html()).toMatch(/Loading/)
        // expect(wrapper.html()).not.toMatch(/Hello/)
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
  describe('onClient: afterMounted', () => {
    describe('without onSSR component', () => {
      const MyComp = () => <div>Hello</div>
      const wrapper = mount(
        <NoSSR>
          <MyComp />
        </NoSSR>
      )
      it('Should show children', () => {
        // expect(wrapper.html()).toMatch(/Hello/)
        expect(wrapper).toMatchSnapshot()
      })
    })
    describe('with onSSR component', () => {
      const MyComp = () => <div>Hello</div>
      const Loading = () => <div>Loading...</div>
      const wrapper = mount(
        <NoSSR onSSR={<Loading />}>
          <MyComp />
        </NoSSR>
      )
      it('Should show children', () => {
        // expect(wrapper.html()).not.toMatch(/Loading/)
        // expect(wrapper.html()).toMatch(/Hello/)
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
