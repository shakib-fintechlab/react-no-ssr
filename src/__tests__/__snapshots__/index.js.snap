// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoSSR onClient: afterMounted with onSSR component Should show children 1`] = `
<NoSSR
  onSSR={<Loading />}
>
  <MyComp>
    <div>
      Hello
    </div>
  </MyComp>
</NoSSR>
`;

exports[`NoSSR onClient: afterMounted without onSSR component Should show children 1`] = `
<NoSSR>
  <MyComp>
    <div>
      Hello
    </div>
  </MyComp>
</NoSSR>
`;

exports[`NoSSR onServer with onSSR component Should show the onSSR component 1`] = `
<div>
  Loading...
</div>
`;

exports[`NoSSR onServer without onSSR components Should not render children 1`] = `null`;
